backend/Brambles/Backend/ASTToLLVM.hs

```diff
@@ -231,48 +247,87 @@ stmtToLLVM (AST.SReturn (_, t) e) = case t of
   AST.TUnit -> LLVM.retVoid  -- TODO: unit vs void?
   _         -> LLVM.ret =<< exprToLLVM e
 
-progToLLVM :: 
+
+funcToLLVM :: 
   ( MonadSymbolTable m n
-  , LLVM.MonadIRBuilder m
+  , ThrowsASTToLLVMError m
+  , LLVM.MonadModuleBuilder m
+  , MonadFix m
+  ) => LLVM.AST.Name -> AST.Func n 'AST.Typed -> m LLVM.AST.Operand
+funcToLLVM n f@AST.Func{..} = mdo
+  (llvmParams, llvmRet) <- paramAndReturn
+
+  f' <- LLVM.function n llvmParams llvmRet $ \args -> withScope $ do
+    _ <- namedBlock entryBlockName
+    zipWithM_ allocate funcParams args
+    LLVM.ret =<< blockToLLVM funcBody
+
+  addSymbol (AST.unVar funcName) f'  -- TODO: recursion requires lookup
+  pure f'
+    where
+      paramAndReturn = case getType f of  -- TODO: not neat
+        AST.TCallable{..} -> pure
+          ( [ (typeToLLVM t, LLVM.NoParameterName) | t <- paramT ]
+          , typeToLLVM returnT
+          )
+        _ -> throwCodegenError
+
+
+moduleToLLVM :: 
+  ( MonadSymbolTable m n
+  , ThrowsASTToLLVMError m
   , LLVM.MonadModuleBuilder m
   , MonadFix m
-  ) => AST.Prog n 'AST.Typed -> m ()
-progToLLVM (AST.Globals _ stmts) = mapM_ stmtToLLVM stmts
+  ) 
+  => AST.Module n 'AST.Typed 
+  -> m ()
+moduleToLLVM AST.Module{..} =  do
+  unless (null moduleGlobals) throwCodegenError  -- TODO
```

Comment:
No support right now for globals, since they must be generated using constant expressions in LLVM